packages/to-json-schema/src/convertSchema.ts

```diff
@@ -0,0 +1,348 @@
+      any,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      any,
+      v.ErrorMessage<v.TupleWithRestIssue> | undefined
+    >
+  | v.LooseTupleSchema<
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      any,
+      v.ErrorMessage<v.LooseTupleIssue> | undefined
+    >
+  | v.StrictTupleSchema<
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      any,
+      v.ErrorMessage<v.StrictTupleIssue> | undefined
+    >
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | v.ArraySchema<any, v.ErrorMessage<v.ArrayIssue> | undefined>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | v.LazySchema<any>;
+
+/**
+ * Schema or pipe type.
+ */
+type SchemaOrPipe =
+  | Schema
+  | v.SchemaWithPipe<
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      [Schema, ...(Schema | v.PipeAction<any, any, v.BaseIssue<unknown>>)[]]
+    >;
+
+/**
+ * Converts any supported Valibot schema to the JSON schema format.
+ *
+ * @param json The JSON schema object.
+ * @param schema The Valibot schema object.
+ * @param context The conversion context.
+ *
+ * @returns The converted JSON schema.
+ */
+export function convertSchema(
+  json: JSONSchema7,
+  schema: SchemaOrPipe,
+  context: Context
+): JSONSchema7 {
+  const config = context?.config;
+
+  const definitionPath = context.definitionsPathMap.get(schema);
+  if (definitionPath) {
+    json.$ref = definitionPath;
+    return json;
+  }
+
+  // If it is schema with pipe, convert each item of pipe
+  if ('pipe' in schema) {
+    for (let index = 0; index < schema.pipe.length; index++) {
+      const action = schema.pipe[index];
+      if (action.kind === 'schema') {
+        if (index > 0) {
+          if (!config?.force) {
+            throw new Error(
+              'A "pipe" with multiple schemas cannot be converted to JSON schema.'
+            );
+          }
+          return json;
+        }
+        const convertedSchema = convertSchema({}, action, context);
+        if (convertedSchema.$ref) {
+          const name = convertedSchema.$ref.split('/')[2];
+          Object.assign(json, context.definitions[name]);
+        } else {
+          Object.assign(json, convertedSchema);
+        }
+      } else {
+        // @ts-expect-error
+        convertAction(json, action, context);
+      }
+    }
+    return json;
+  }
+
+  // Otherwise, convert individual schema to JSON schema
+  switch (schema.type) {
+    case 'any':
+    case 'unknown': {
+      break;
+    }
+
+    case 'null': {
+      json.type = 'null';
+      break;
+    }
+
+    case 'nullable':
+    case 'nullish': {
+      json.anyOf = [
+        convertSchema({}, schema.wrapped as SchemaOrPipe, context),
+        { type: 'null' },
+      ];
+      const defaultValue = v.getDefault(schema);
+      if (
+        defaultValue !== undefined &&
+        assertJSON(
```

Comment:
Is it even necessary to check if the default value is JSON compatible? Valibot already forces the user to match the value to the input type of the schema, and any invalid schema will result in an error anyway. The only expectation is values like `NaN` or `Infinity`, but that's a rare case I think we can ignore for now.

Reply:
But the default value here can be anything right ? It could be a Map, a Set, a Date, a File, etc.
All of which would break in JSON 🤔

Reply:
Ah but the wrapped schema would be incompatible.
Then yes it would only be for fringe cases. We could remove 👍